api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -228,21 +228,32 @@ private void ensureDrugOrderAutoExpirationDateIsSet(Order order) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
 	}
-
+	
 	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
 		if (order.getOrderType() != null) {
 			return;
 		}
+		List<OrderType>orderTypes = new ArrayList<>();
 		OrderType orderType = null;
 		if (orderContext != null) {
 			orderType = orderContext.getOrderType();
 		}
 		if (orderType == null) {
 			orderType = getOrderTypeByConcept(order.getConcept());
 		}
+		if((orderTypes.size() == 1)) {
```

Comment:
Did take time to carefully review your changes?

Reply:
yes i did . However i do apologise , the double Braces was not very obvious for me i have just realised it now.